### PR TITLE
IAM policy for Azure VM Scale Set in k8s

### PR DESCRIPTION
**Pull request by robgil-dds**
This changes the way k8s is identified with regard to its identifying principal. Before we were using User Assigned. This moves to SystemAssigned. With a SystemAssigned service principal, we can grant all the vms in the scale set access to the keyvault. 

This should potentially change to pod based in the future to have more granular control over what secrets an application has access to for increased security. 

The SystemAssigned managed identity requires a preview feature to be
enabled.

```
rgil@rem5:~/atst/terraform/providers/dev$ az feature list|grep MSIPreview
    "id": "/subscriptions/95934d54-980d-47cc-9bce-3a96bf9a2d1b/providers/Microsoft.Features/providers/Microsoft.ContainerService/features/MSIPreview",
    "name": "Microsoft.ContainerService/MSIPreview",
rgil@rem5:~/atst/terraform/providers/dev$ az feature register --namespace Microsoft.ContainerService --name MSIPreview
Once the feature 'MSIPreview' is registered, invoking 'az provider register -n Microsoft.ContainerService' is required to get the change propagated
{
  "id": "/subscriptions/95934d54-980d-47cc-9bce-3a96bf9a2d1b/providers/Microsoft.Features/providers/Microsoft.ContainerService/features/MSIPreview",
  "name": "Microsoft.ContainerService/MSIPreview",
  "properties": {
    "state": "Registering"
  },
  "type": "Microsoft.Features/providers/features"
}
rgil@rem5:~/atst/terraform/providers/dev$ az provider register -n Microsoft.ContainerService
rgil@rem5:~/atst/terraform/providers/dev$
```

This also now integrates the policy for keyvault with the k8s managed
identity (system assigned).

### CODE REVIEW

File: terraform/modules/k8s/main.tf [hunk 1]
```diff
@@ -25,6 +25,9 @@ resource "azurerm_kubernetes_cluster" "k8s" {
     min_count             = var.min_count # FIXME: if auto_scaling disabled, set to 0
   }
 
+  identity {
+    type = "SystemAssigned"
+  }
```
Comment (dandds) [inline terraform/modules/k8s/main.tf]: `terraform plan` says this is applied, but when I query for info about the identity associated with the VMSS, I get the following:

```
❯ az vmss identity show -g MC_cloudzero-dev-vpc_cloudzero-dev-k8s_eastus2 -n aks-default-54410534-vmss
{
  "principalId": null,
  "tenantId": null,
  "type": "UserAssigned",
  "userAssignedIdentities": {
    "/subscriptions/95934d54-980d-47cc-9bce-3a96bf9a2d1b/resourceGroups/MC_cloudzero-dev-vpc_cloudzero-dev-k8s_eastus2/providers/Microsoft.ManagedIdentity/userAssignedIdentities/cloudzero-dev-k8s-agentpool": {
      "clientId": "19affc7a-027f-48ac-9c21-ccc0a94cbe33",
      "principalId": "d0795d2b-5aa7-442c-8978-29f9a7390475"
    }
  }
}
```

(i.e., "UserAssigned") Do we need to destroy and rebuild to get the system-assigned identity?

Comment (dandds) [reply terraform/modules/k8s/main.tf]: There's not... another identity associated with the cluster, is there?

**Comment by dandds** [comment]
Oh also, could you document this:

> az provider register -n Microsoft.ContainerService

in the TF README?

File: terraform/modules/k8s/main.tf (hunk 1 above)
Comment (robgil-dds) [reply terraform/modules/k8s/main.tf]: Strange. I see that now also. That should show as SystemAssigned but its not. I'll have to troubleshoot.

**Comment by robgil-dds** [comment]
@dandds Updated the docs with the Preview feature registration steps.

**Comment by robgil-dds** [comment]
See https://github.com/terraform-providers/terraform-provider-azurerm/issues/5388

**Pull request merged**